### PR TITLE
[serve] Add multi app support to `serve run`

### PR DESCRIPTION
**Pull request by zcin**
## Why are these changes needed?

Add support for `serve run` to run multiple applications at once.

To run multiple applications with a config file:
```
serve run new_config
```

CANNOT run multiple applications with import path arguments:

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by zcin** [comment]
@shrekris-anyscale @sihanwang41 @edoakes PTAL!

File: python/ray/serve/scripts.py [hunk 1]
```diff
@@ -298,7 +298,7 @@ def deploy(config_file_name: str, address: str):
     ),
 )
 def run(
-    config_or_import_path: str,
+    config_or_import_path: Tuple[str],
```
Comment (sihanwang41) [inline python/ray/serve/scripts.py:301]: why do need to be a Tuple?

File: python/ray/serve/tests/test_cli.py [hunk 2]
```diff
@@ -667,6 +667,43 @@ def test_run_application(ray_start_stop):
...
+@pytest.mark.skipif(sys.platform == "win32", reason="File path incorrect on Windows.")
+def test_run_multi_app(ray_start_stop):
+    """Deploys valid multi-app config file via `serve run`."""
+
+    # Deploy via config file
+    config_file_name = os.path.join(
+        os.path.dirname(__file__), "test_config_files", "pizza_world.yaml"
+    )
+
+    print('Running config file "arithmetic.yaml".')
```
Comment (sihanwang41) [inline python/ray/serve/tests/test_cli.py:679]: nit: pizza_world.yaml

File: python/ray/serve/tests/test_cli.py [hunk 3]
```diff
@@ -667,6 +667,43 @@ def test_run_application(ray_start_stop):
     print("Kill successful! Deployment is not reachable over HTTP.")
 
 
+@pytest.mark.skipif(sys.platform == "win32", reason="File path incorrect on Windows.")
```
Comment (sihanwang41) [inline python/ray/serve/tests/test_cli.py:670]: Add a simple failure test case when you send ill-format yaml file.

File: python/ray/serve/scripts.py (hunk 1 above)
Comment (zcin) [reply python/ray/serve/scripts.py]: A mistake, sorry, removed

File: python/ray/serve/tests/test_cli.py (hunk 3 above)
Comment (zcin) [reply python/ray/serve/tests/test_cli.py:686]: Added!

**Review by edoakes — APPROVED** [review]

**Comment by zcin** [comment]
@edoakes should be ready for merge

**Pull request merged**